unittests/forked_tests_if.cpp

```diff
@@ -153,17 +188,530 @@ BOOST_FIXTURE_TEST_CASE(fork_with_bad_block_if, savanna_cluster::cluster_t) try
+      _nodes[0].produce_block();
+   }
+   const vector<account_name> producers { "dan"_n, "sam"_n, "pam"_n };
+   _nodes[0].create_accounts(producers);
+   auto prod = set_producers(0, producers);   // set new producers and produce blocks until the switch is pending
+
+   auto sb_common = _nodes[0].produce_blocks(4);
+   auto lib = _nodes[0].lib_num();
+   BOOST_REQUIRE_EQUAL(sb_common->producer, producers[prod]); // first block produced by producers[prod]
+
+
+   const std::vector<size_t> partition {1, 2, 3};
+   set_partition(partition);                  // simulate 2 disconnected partitions:
+                                              // P0 (node {0}) and P1 (nodes {1, 2, 3}).
+                                              // At this point, each node has a QC to include into
+                                              // the next block it produces which will advance lib by one)
+                                              // finality will still advance further in p1 because it has 3 finalizers
+
+   _nodes[1].produce_blocks(2);                   // produce 2 blocks on _nodes[1] finality will advance by 2 blocks
+   auto node1_head = _nodes[1].head();
+   BOOST_REQUIRE_EQUAL(_nodes[1].lib_num(), lib+2);
+
+   _nodes[0].produce_block(_block_interval_us * 12); // produce 2 blocks on _nodes[0]. finality will advance by 1 block only
+   _nodes[0].produce_block();                        // but they'll have a later timestamp
+   auto node0_head = _nodes[0].head();
+   BOOST_REQUIRE_EQUAL(_nodes[0].lib_num(), lib+1);
+
+   // verify assumptions (finality more advanced on _nodes[1], but timestamp less)
+   auto core0 = block_handle_accessor::core_info(node0_head);
+   auto core1 = block_handle_accessor::core_info(node1_head);
+   BOOST_REQUIRE_GT(core1->last_final_block_num, core0->last_final_block_num);
+   BOOST_REQUIRE_GT(core1->last_qc_block_num, core0->last_qc_block_num);
+   BOOST_REQUIRE_LT(core1->timestamp, core0->timestamp);
+
+   BOOST_REQUIRE_EQUAL(_nodes[0].head().id(), node0_head.id());
+
+   push_blocks(1, 0, sb_common->block_num() + 1); // push the 2 produced blocks to _nodes[0]
+   BOOST_REQUIRE_EQUAL(_nodes[0].head().id(), node1_head.id()); // and check that we fork-switched to _nodes[1]'s head
+
+   set_partition({});
+   propagate_heads();
+   verify_lib_advances();
+
+} FC_LOG_AND_RETHROW()
+
+
+// ---------------------------- irreversible_mode_if_1 ---------------------------------
+//   A node not in irreversible mode captures what the LIB is as of different blocks
+//   in the blockchain. Then the blocks are synced to a node running in irreversible
+//   mode gradually. When syncing up to some block number, the test checks that the
+//   controller head is at the block number of the corresponding LIB that was captured
+//   earlier.
+// -------------------------------------------------------------------------------------
+BOOST_FIXTURE_TEST_CASE( irreversible_mode_if_1, savanna_cluster::cluster_t ) try {
```

Comment:
Rename to `irreversible_mode_savanna_1`, also comment above and `_2` below.

Reply:
done